[WIP]ChatSpaceのDB設計

kouhe0115 opened the pull request:
DB設計しました
Add users table
Add groups table
Add messages table
Add groups_users table
Add users association
Add groups association
Add messages association
Add groups_users association

kouhe0115 committed:
Edit README.md

Add users table
Add groups table
Add messages table
Add groups_users table
Add users association
Add groups association
Add messages association
Add groups_users association

kouhe0115 commented:
## What
DBのテーブル、カラム、アソシエーションの設計しました

## Why
事前にしっかりと設計をしておかないと、後から変更が難儀なため。
一番基盤の設計であります

yuki-uchida commented on README.md:
```diff
@@ -1,24 +1,69 @@
-* Deployment instructions
+| column     | type   | option   |
+|------------|--------|----------|
+| message_id | int    |          |
```
idでよいです。また、idは全てのテーブルでデフォルトで実装されるのでなくてよいです

yuki-uchida commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| column   | type            | option   |
+|----------|-----------------|----------|
+| id       | int             |          |
+| user_id  | refference      | not null |
```
null： falseが正しい書き方です

kouhe0115 committed:
Edit.README.md

Users table id DELETE
Messages table id DELETE
Groups table id DELETE

kouhe0115 committed:
Edit.READEM,md

Users table id delete
Messages table id delete
Groups table id delete
not null → Null:false

kouhe0115 commented:
レビューありがとうございます。
上記ご指摘箇所、修正しました。
ご確認お願いします。

yuki-uchida commented on README.md:
```diff
@@ -1,24 +1,66 @@
+|------------|--------|--------------|
+| body       | text   |              |
+| image      | string |              |
+| user_id    | int    | null : false |
```
外部キーの場合にはrefference型にしてforeign_key: trueとする必要があります

kouhe0115 committed:
Edit.README.md

type int→ refference
option Add FOREIGN KEY

kouhe0115 commented:
上記指摘点修正しました。
ご確認お願いします。

yuki-uchida commented:
LGTMです。お疲れ様でした

kouhe0115 commented:
ありがとうございます。
